packages/cli/src/ConcurrencyQueue.ts

```diff
@@ -0,0 +1,30 @@
+export class ConcurrencyQueue {
+	private waiting: Array<[string, () => void]> = [];
+
+	constructor(private capacity: number) {}
+
+	async enqueue(id: string) {
```

Comment:
This merits a docline? `Pause the execution if it exceeds max concurrency capacity`?